lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -106,7 +106,18 @@ def self.config_exist?(config_root)
   def self.symkeys_to_string(struct)
     case(struct)
     when Hash
-      Hash[struct.map { |k,v| [k.to_s, symkeys_to_string(v)] }]
+      map = struct.map { |k,v|
+        case k.class.to_s
+        when "Symbol"
+          key = k.to_s
+        else
+          key = k
+        end
+        value = symkeys_to_string(v)
+        [key, value]
+      }
+
+      Hash[map]
```

Comment:
It does affect data as well. It is called from the `#yaml_data` and `#eyaml_lookup_key` functions and I suspect that if we remove those calls, then we will start seeing problems with users that have symbolic keys in their data. That's just a guess though. The change, optimized as suggested by @hlindberg, should be fine in any case. Perhaps we should consider removing the calls from `#yaml_data` and `#eyaml_lookup_key` too?